web-bluetooth/read-characteristic-value-changed.js

```diff
@@ -1,13 +1,38 @@
+var bluetoothDevice;
 var batteryLevelCharacteristic;
 
 function onReadBatteryLevelButtonClick() {
-  log('Requesting Bluetooth Device...');
-  navigator.bluetooth.requestDevice(
-    {filters: anyDevice(), optionalServices: ['battery_service']})
-  .then(device => {
-    log('Connecting to GATT Server...');
-    return device.gatt.connect();
+  requestDevice()
+  .then(connectDeviceAndCacheCharacteristics)
+  .then(_ => {
```

Comment:
[`BluetoothRemoteGATTCharacteristic.writeValue`](https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-writevalue) also resolves promise with `undefined` but we still use `_` in https://googlechrome.github.io/samples/web-bluetooth/reset-energy.html.
I'll go ahead and merge this PR.
Please open a separate issue for considering `()` for `undefined` values.